modules/src/core/ics04_channel/timeout.rs

```diff
@@ -0,0 +1,124 @@
+use crate::prelude::*;
+
+use core::fmt::Display;
+
+use crate::{core::ics02_client::error::Error as ICS2Error, Height};
+
+use ibc_proto::ibc::core::client::v1::Height as RawHeight;
+use serde::{Deserialize, Serialize};
+use tendermint::abci::tag::Value as TagValue;
+
+/// Represents a height to be used to timeout packets.
+///
+/// `TimeoutHeight` is treated differently from other heights because
+///
+/// `RawHeight.timeout_height == {revision_number: 0, revision_height = 0}`
+///
+/// is legal and meaningful, even though the Tendermint spec rejects this
+/// height as invalid. Thus, it must be parsed specially, where this
+/// special case means "no timeout".
+#[derive(Clone, Copy, Debug, Default, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct TimeoutHeight(Option<Height>);
+
+impl TimeoutHeight {
+    pub fn no_timeout() -> Self {
+        Self(None)
+    }
+
+    /// Revision number to be used in packet commitment computation
+    pub fn commitment_revision_number(&self) -> u64 {
+        match self.0 {
+            Some(height) => height.revision_number(),
+            None => 0,
+        }
+    }
+
+    /// Revision height to be used in packet commitment computation
+    pub fn commitment_revision_height(&self) -> u64 {
+        match self.0 {
+            Some(height) => height.revision_height(),
+            None => 0,
+        }
+    }
+
+    /// Check if a height is past the timeout height, and thus is expired.
+    pub fn has_expired(&self, height: &Height) -> bool {
+        match self.0 {
+            Some(timeout_height) => *height >= timeout_height,
+            // When there's no timeout, heights are never expired
+            None => false,
+        }
+    }
+
+    // FIXME: Find better name
+    pub fn has_timeout(&self) -> bool {
```

Comment:
I find the `has_timeout()` name confusing, how about `fn never_expires() -> bool` because that is what we want to check elsewhere in the code. `TimeoutHeight` could also be modelled like so ->
```rust
pub enum TimeoutHeight {
  Never,
  At(Height)
}
```

Reply:
Seems like we wouldn't need the `has_timeout()` function at all if we fix the bug I described earlier (https://github.com/informalsystems/ibc-rs/pull/2346/files#r910757508).

Reply:
Good idea!

[95eb992](https://github.com/informalsystems/ibc-rs/pull/2346/commits/95eb992deb7a32d06c44b41215601aff9311f3f8)